### PR TITLE
[Feature] Instance Group Memberships

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000
https://jiracloud.swg.usma.ibm.com:8443/browse/UI-16121
https://jiracloud.swg.usma.ibm.com:8443/browse/UI-16382

Output from acceptance testing:

Screenshot 2021-04-28 at 2 57 29 PM

Screenshot 2021-05-12 at 6 06 30 PM
Screenshot 2021-05-12 at 5 59 22 PM

Screenshot 2021-05-12 at 5 49 54 PM

```
$ make testacc TEST=./ibm TESTARGS='-run=TestAccIbmIsInstanceGroupMembershipResource_Basic'
$ make testacc TEST=./ibm TESTARGS='-run=TestAccIbmIsInstanceGroupMembershipDataSource_Basic' 
$ make testacc TEST=./ibm TESTARGS='-run=TestAccIbmIsInstanceGroupMembershipsDataSource_Basic' 

```

### CODE REVIEW

File: ibm/data_source_ibm_is_instance_group_memberships.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		membership := map[string]interface{}{
+			"delete_instance_on_membership_delete": *instanceGroupMembership.DeleteInstanceOnMembershipDelete,
+			"instance_group_membership":            *instanceGroupMembership.ID,
+			"name":                                 *instanceGroupMembership.Name,
+			"status":                               *instanceGroupMembership.Status,
+		}
+
+		instances := make([]map[string]interface{}, 0)
+		if instanceGroupMembership.Instance != nil {
+			log.Println("*instanceGroupMembership.Instance.CRN - Instance")
```
Comment (hkantare) [inline ibm/data_source_ibm_is_instance_group_memberships.go:158]: Remove log stmts which are not required

Comment (SunithaGudisagarIBM) [reply ibm/data_source_ibm_is_instance_group_memberships.go]: @hkantare Removed Logs, Can you please review and merge the PR..

File: ibm/resource_ibm_is_instance_group_membership.go [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+	}
+	instanceGroupMembership, response, err := sess.GetInstanceGroupMembership(&getInstanceGroupMembershipOptions)
+	if err != nil || instanceGroupMembership == nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error Getting InstanceGroup Membership: %s\n%s", err, response)
+	}
+	d.Set("delete_instance_on_membership_delete", *instanceGroupMembership.DeleteInstanceOnMembershipDelete)
```
Comment (VaishnaviGopal) [inline ibm/resource_ibm_is_instance_group_membership.go:225]: add d.setid in read method. If user passes only ins grp id and membership as fields to the resource, the tfstate should contain the details of that resource along with an id.

**Review by VaishnaviGopal — CHANGES_REQUESTED** [review]

**Review by VaishnaviGopal — COMMENTED** [review]
Merge all 5 commits to 1

**Review by VaishnaviGopal — APPROVED** [review]

File: ibm/resource_ibm_is_instance_group_membership.go [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+		Read:     resourceIBMISInstanceGroupMembershipRead,
+		Update:   resourceIBMISInstanceGroupMembershipUpdate,
+		Delete:   resourceIBMISInstanceGroupMembershipDelete,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+
+			"instance_group": {
+				Type:         schema.TypeString,
+				Required:     true,
```
Comment (hkantare) [inline ibm/resource_ibm_is_instance_group_membership.go:25]: Does instance_group needs to be force new?

File: website/docs/d/is_instance_group_membership.html.markdown [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+layout: "ibm"
+page_title: "IBM : is_instance_group_membership"
+description: |-
+  Get information about InstanceGroupMembership
+subcategory: "Virtual Private Cloud API"
+---
+
+# ibm\_is_instance_group_membership
+
+Retrives instance group memership info of an instance group
```
Comment (hkantare) [inline website/docs/d/is_instance_group_membership.html.markdown:11]: Retrives spelling

File: website/docs/d/is_instance_group_membership.html.markdown [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Example Usage
+
+```hcl
+data "is_instance_group_membership" "is_instance_group_membership" {
+	instance_group = "r006-76740f94-fcc4-11e9-96e7-a77723715315"
```
Comment (hkantare) [inline website/docs/d/is_instance_group_membership.html.markdown:17]: format the example

File: website/docs/r/is_instance_group_membership.html.markdown [hunk 6]
```diff
@@ -0,0 +1,54 @@
+---
+layout: "ibm"
+page_title: "IBM : is_instance_group_membership"
+description: |-
+  Manages Instance Group Membership
+subcategory: "Virtual Private Cloud API"
```
Comment (hkantare) [inline website/docs/r/is_instance_group_membership.html.markdown:6]: The subcategory should be "VPC infrastructure" in all docs and keep it at top before layout

File: ibm/data_source_ibm_is_instance_group_memberships.go [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"crn": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The CRN for this instance template.",
+									},
+									"instance_template": {
```
Comment (astha-jain) [inline ibm/data_source_ibm_is_instance_group_memberships.go:76]: Please use constants for property names

File: ibm/resource_ibm_is_instance_group_membership.go [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+	instanceGroupMembershipID := d.Get(isInstanceGroupMembership).(string)
+
+	getInstanceGroupMembershipOptions := vpcv1.GetInstanceGroupMembershipOptions{
+		ID:              &instanceGroupMembershipID,
+		InstanceGroupID: &instanceGroupID,
+	}
+
+	instanceGroupMembership, response, err := sess.GetInstanceGroupMembership(&getInstanceGroupMembershipOptions)
+	if err != nil || instanceGroupMembership == nil {
+		if response != nil && response.StatusCode == 404 {
```
Comment (hkantare) [inline ibm/resource_ibm_is_instance_group_membership.go:179]: Why are returning nil if its 404 ? just retrun the error

File: website/docs/d/is_instance_group_membership.html.markdown [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+# ibm\_is_instance_group_membership
+
+Retrieves instance group memership info of an instance group
+
+## Example Usage
+
+```
+data "is_instance_group_membership" "is_instance_group_membership" {
```
Comment (hkantare) [inline website/docs/d/is_instance_group_membership.html.markdown:16]: format the doc example

**Pull request merged**